apps/newsletters-ui/src/app/routes/newsletter.tsx

```diff
@@ -1,22 +1,35 @@
 import type { LoaderFunction, RouteObject } from 'react-router-dom';
-import type { Newsletter } from '@newsletters-nx/newsletters-data-client';
+import type {
+	ApiResponse,
+	Newsletter,
+} from '@newsletters-nx/newsletters-data-client';
 import { NewsletterDetailView } from '../components/NewsletterDetailView';
 import { NewsletterListView } from '../components/NewslettersListView';
 import { ErrorPage } from '../ErrorPage';
 import { Layout } from '../Layout';
 
-export async function getNewsletters(): Promise<Newsletter[]> {
+async function getNewsletters(): Promise<Newsletter[]> {
 	try {
 		const response = await fetch('api/v1/newsletters');
-		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- internal api call
-		const data = await response.json();
-		return data as Newsletter[];
+		const data = (await response.json()) as ApiResponse<Newsletter[]>;
+		return data.results;
 	} catch (err) {
 		console.error(err);
 		return [];
 	}
 }
 
+async function getNewsletter(id: string): Promise<Newsletter | undefined> {
+	try {
+		const response = await fetch(`api/v1/newsletters/${id}`);
+		const data = (await response.json()) as ApiResponse<Newsletter>;
+		return data.results;
```

Comment:
The results would always be `undefined` for for an error response, but it would be safer to check the `ok`. The return value of the function is `Promise<Newsletter | undefined>` , and using `data.ok` could return `false`, but that's fixable.

Looking at this again, I think it would be clearer to define `ApiErrorRepsonse` as an explict type and change the parsed statement to something like:
``` typescript
const data = (await response.json()) as ApiResponse<Newsletter> | ApiErrorRepsonse;
```